src/main/java/com/courtcanva/ccfranchise/exceptions/ControllerExceptionHandler.java

```diff
@@ -0,0 +1,59 @@
+package com.courtcanva.ccfranchise.exceptions;
+
+import com.courtcanva.ccfranchise.dtos.ErrorDto;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.support.DefaultMessageSourceResolvable;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@ControllerAdvice
+public class ControllerExceptionHandler {
+
+    @ExceptionHandler(ResourceNotFoundException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public ErrorDto handleResourceNotFoundException(ResourceNotFoundException e) {
+
+        log.error(e.getMessage(), e);
+        return ErrorDto.builder()
+                .error(HttpStatus.NOT_FOUND.getReasonPhrase())
+                .details(List.of(e.getMessage()))
```

Comment:
Don't directly return the exception message as a response, it would be too technical sometimes.

Reply:
thx, I add description message to the ErrorDto